core/src/main/kotlin/entity/interaction/ContextInteraction.kt

```diff
@@ -0,0 +1,119 @@
+package dev.kord.core.entity.interaction
+
+import dev.kord.core.Kord
+import dev.kord.core.behavior.UserBehavior
+import dev.kord.core.behavior.interaction.ApplicationCommandInteractionBehavior
+import dev.kord.core.cache.data.InteractionData
+import dev.kord.core.entity.Member
+import dev.kord.core.entity.Message
+import dev.kord.core.entity.User
+import dev.kord.core.supplier.EntitySupplier
+import dev.kord.core.supplier.EntitySupplyStrategy
+
+/**
+ * Represents an interaction of type [ApplicationCommand][dev.kord.common.entity.InteractionType.ApplicationCommand]
+ */
+sealed interface ApplicationCommandInteraction : Interaction, ApplicationCommandInteractionBehavior
+
+
+/**
+ * A [ApplicationCommandInteraction] that's invoked through chat input.
+ */
+sealed interface  ChatInputCommandInteraction : ApplicationCommandInteraction {
+    val command: InteractionCommand get() =  InteractionCommand(data.data, kord)
+
+
+}
+
+/**
+ * A [ApplicationCommandInteraction] that's invoked through chat input specific to a guild.
+ */
+class GuildChatInputCommandInteraction(
+    override val data: InteractionData,
+    override val kord: Kord,
+    override val supplier: EntitySupplier
+) : ChatInputCommandInteraction, GuildApplicationCommandInteraction
+
+
+
+/**
+ * A [ApplicationCommandInteraction] that's invoked through chat input.
+ */
+class GlobalChatInputCommandInteraction(
+    override val data: InteractionData,
+    override val kord: Kord,
+    override val supplier: EntitySupplier
+) : ChatInputCommandInteraction, GlobalApplicationCommandInteraction
+
+
+/**
+ * A [ApplicationCommandInteraction] that's invoked through user commands.
+ */
+sealed interface  UserCommandInteraction : ApplicationCommandInteraction {
```

Comment:
The only way to access the command id/name seems to be via `interaction.data.data.name.value` (same goes for Message commands)

Chat input commands have `interaction.command.rootName`

Reply:
that's not the thing you are looking for.
you can solve this by casting you have three types
RootCommand, SubCommand, GroupSubCommand.

just do ```kt
val command = interaction.command
if(command is SubCommand) ...
```

Reply:
`UserCommandInteraction` _does not have_ a `command` field - my issue is not with slash commands, but with the other 2 types

Reply:
You were right, I missed that one, now it's included in both interactions and it's derivaites

Reply:
can we get it for the id field too? (`Interaction.data.data.id`)
I match mine with their handlers by snowflake